### PR TITLE
Use `walk!` in bacterial growth example

### LINKED ISSUE #544: Out of bounds error ContinuousSpace when adding agents
**Describe the bug**
I am trying to model a system of of growing bacteria in continuous space, cfr. [bacteria growth](https://juliadynamics.github.io/Agents.jl/stable/examples/growing_bacteria/). It seems that when the number of agents is too large, I have an out of bounds error due to the discretisation of the space. It is unclear to me why this should occur, as I understand that this is only to speed up the neighbour search. 

Either this is a true bug, or it is something that needs some additional documentation.

```julia
ERROR: LoadError: BoundsError: attempt to access 200×200 Matrix{Vector{Int64}} at index [22, 0]
Stacktrace:
  [1] getindex
    @ ./array.jl:802 [inlined]
  [2] add_agent_to_space!
    @ ~/.julia/packages/Agents/uOpQY/src/spaces/continuous.jl:77 [inlined]
  [3] add_agent_pos!(agent::Agent, model::AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister})
    @ Agents ~/.julia/packages/Agents/uOpQY/src/core/space_interaction_API.jl:213
  [4] #add_agent!#22
    @ ~/.julia/packages/Agents/uOpQY/src/core/space_interaction_API.jl:298 [inlined]
  [5] add_agent!
    @ ~/.julia/packages/Agents/uOpQY/src/core/space_interaction_API.jl:296 [inlined]
  [6] #add_agent!#21
    @ ~/.julia/packages/Agents/uOpQY/src/core/space_interaction_API.jl:285 [inlined]
  [7] add_agent!
    @ ~/.julia/packages/Agents/uOpQY/src/core/space_interaction_API.jl:285 [inlined]
  [8] agent_step!(agent::Agent, model::AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister})
    @ Main ~/SVN_GITHUB/Personal_git/PhageSim/scripts/bacteria_growth/continuous.jl:68
  [9] step!(model::AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister}, agent_step!::typeof(agent_step!), model_step!::typeof(dummystep), n::Int64, agents_first::Bool)
    @ Agents ~/.julia/packages/Agents/uOpQY/src/simulations/step.jl:55
 [10] step!(model::AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister}, agent_step!::Function, model_step!::Function, n::Int64)
    @ Agents ~/.julia/packages/Agents/uOpQY/src/simulations/step.jl:48
 [11] step!(abmstepper::InteractiveDynamics.ABMStepper, model::AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister}, agent_step!::Function, model_step!::Function, n::Int64)
    @ InteractiveDynamics ~/.julia/packages/InteractiveDynamics/Ura10/src/agents/stepping.jl:153
 [12] (::InteractiveDynamics.var"#66#69"{Int64, Int64, AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister}, typeof(agent_step!), typeof(dummystep), InteractiveDynamics.ABMStepper, Observable{Int64}})(io::VideoStream)
    @ InteractiveDynamics ~/.julia/packages/InteractiveDynamics/Ura10/src/agents/plots_videos.jl:219
 [13] #Record#937
    @ ~/.julia/packages/Makie/xbI6d/src/display.jl:586 [inlined]
 [14] #record#936
    @ ~/.julia/packages/Makie/xbI6d/src/display.jl:580 [inlined]
 [15] abm_video(file::String, model::AgentBasedModel{ContinuousSpace{2, true, Float64, typeof(Agents.defvel)}, Agent, typeof(Agents.Schedulers.fastest), Dict{Symbol, Float64}, MersenneTwister}, agent_step!::typeof(agent_step!), model_step!::typeof(dummystep); spf::Int64, framerate::Int64, frames::Int64, resolution::Tuple{Int64, Int64}, title::String, showstep::Bool, axiskwargs::NamedTuple{(), Tuple{}}, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ InteractiveDynamics ~/.julia/packages/InteractiveDynamics/Ura10/src/agents/plots_videos.jl:216
 [16] top-level scope
    @ ~/SVN_GITHUB/Personal_git/PhageSim/scripts/bacteria_growth/continuous.jl:116
in expression starting at /Users/michielstock/SVN_GITHUB/Personal_git/PhageSim/scripts/bacteria_growth/continuous.jl:116
```

**Minimal Working Example**

Just run the bacterial growth [example] a bit longer to reproduce this bug: 

```julia
abm_video(
    "bacteria.mp4", model, agent_step!, model_step!;
    am = cassini_oval, ac = bacteria_color,
    spf = 50, framerate = 30, frames = 2000,
    title = "Growing bacteria"
)
```

My code is a bit more similar to the continuous Covid agents model, though with reproducing agents.

**Agents.jl version**

v4.5.1

### PR DESCRIPTION
**Pull request by MichielStock**
Addresses #544, though does not solve it.

Closes #544

### CODE REVIEW

**Comment by AayushSabharwal** [comment]
I think this looks good. Maybe address the `add_agent!` issue here too?

**Comment by Datseris** [comment]
bump @MichielStock can you address the `add_agent!` here as Aayush suggested?

**Comment by MichielStock** [comment]
I added a check now. It runs with an unlimited number of bacteria now, as they now completely remain in the space.

File: examples/growing_bacteria.jl [hunk 1]
```diff
@@ -65,12 +65,13 @@ nothing # hide
...
 function model_step!(model)
+    extent = model.space.extent
     for a in allagents(model)
         if a.growthprog ≥ 1
             ## When a cell has matured, it divides into two daughter cells on the
-            ## positions of its nodes.
-            add_agent!(a.p1, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
-            add_agent!(a.p2, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
+            ## positions of its nodes. Only place a cell when it is in the extend
+            all(0 .≤ a.p1 .≤ extent) && add_agent!(a.p1, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
```
Comment (Datseris) [inline examples/growing_bacteria.jl:73]:
```suggestion
            all(0 .≤ a.p1 .< extent) && add_agent!(a.p1, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
```
As discussed in the issue, if the position is _equal_ to the extent, it is out of bounds.

**Review by Datseris — CHANGES_REQUESTED** [review]

File: examples/growing_bacteria.jl [hunk 2]
```diff
@@ -70,8 +70,8 @@ function model_step!(model)
         if a.growthprog ≥ 1
             ## When a cell has matured, it divides into two daughter cells on the
             ## positions of its nodes. Only place a cell when it is in the extend
-            all(0 .≤ a.p1 .≤ extent) && add_agent!(a.p1, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
-            all(0 .≤ a.p2 .≤ extent) && add_agent!(a.p2, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
+            all(0 .< a.p1 .< extent) && add_agent!(a.p1, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
+            all(0 .< a.p2 .< extent) && add_agent!(a.p2, model, 0.0, a.orientation, 0.0, 0.1 * rand(model.rng) + 0.05)
```
Comment (Datseris) [inline examples/growing_bacteria.jl:74]: The 0 is allowed. you can have it equal to 0, just not equal to extent.

**Review by Datseris — APPROVED** [review]

**Comment by Datseris** [comment]
@MichielStock thank you! In Agents.jl 5.0 all examples besides the 5 central ones to the library will be in a new repository called "Agents.jl Example Zoo". Can you please make this PR there instead? https://github.com/JuliaDynamics/Agents.jl_Example_Zoo

(please close this when you open the new PR)

**Pull request closed without merging**